src/components/dialog.js

```diff
@@ -230,15 +233,45 @@ export class Dialog {
   }
 
   /**
-   * Whether to display loading indicator.
-   * @param {boolean} isLoading
+   * Transitions to the next view
+   * @param {!./view.View} view
    */
-  setLoading(isLoading) {
-    if (isLoading) {
+  entryTransitionToNextView(view) {
+    // Temporarily cache the old view
+    this.previousView_ = this.view_;
```

Comment:
As a small variation, you could store `this.previousView_` _only_ when it has a progress indicator. You could even call it `this.prevProgressView_` to make it clear. That way some conditions further down would be a bit simpler.